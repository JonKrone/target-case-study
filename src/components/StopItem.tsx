import React from 'react'

interface StopItemProps {
  item: NextTrip.Stop
}

export const StopItem: React.FC<StopItemProps> = ({ item: { Text } }) => {
  return <li style={{ paddingTop: '0.5rem' }}>{Text}</li>
}

export default StopItem
