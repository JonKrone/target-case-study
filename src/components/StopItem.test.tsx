import React from 'react'
import { render } from '@testing-library/react'

import StopItem from './StopItem'

const testStop = { Text: 'P. Sherman 42 Wallaby Way, Sydney', Value: '1' }

describe('StopItem', () => {
  it('renders', () => {
    const { asFragment, getByText } = render(<StopItem item={testStop} />)

    expect(getByText(/Wallaby/)).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
