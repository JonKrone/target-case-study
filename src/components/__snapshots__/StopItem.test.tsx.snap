// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopItem renders 1`] = `
<DocumentFragment>
  <li
    style="padding-top: 0.5rem;"
  >
    P. Sherman 42 Wallaby Way, Sydney
  </li>
</DocumentFragment>
`;
